src/components/Button/Button.tsx

```diff
@@ -195,11 +201,21 @@ export default class Button extends React.Component<ButtonProps> {
 
     const className = this.isLoading() ? 'y-button--content__hidden' : '';
 
+    const leftIconColumn = iconPosition === 'left' &&
+      buttonIcon && <FixedGridColumn fixed={true}>{buttonIcon}</FixedGridColumn>;
+    const rightIconColumn = iconPosition === 'right' &&
+      buttonIcon && <FixedGridColumn fixed={true}>{buttonIcon}</FixedGridColumn>;
+
     return (
       <Block textSize={textSize} className={className}>
-        {iconPosition === 'left' && buttonIcon}
-        {text}
-        {iconPosition === 'right' && buttonIcon}
+        <FixedGridRow gutterSize={GutterSize.NONE}>
+          {leftIconColumn}
+          <FixedGridColumn>
```

Comment:
i was thinking this `FixedGridColumn` would also need to be `fixed` to prevent it from pushing out wider, but if the outer `display:flex Button` prevents that then we're good. may want to double check in other browsers like IE/Edge